File: frontend/src/model/SchoolClass.ts
import {UserView} from "./User";

export interface SchoolClass {
    id: number,
    name: string,
    mainTeacher: UserView,
    schoolId: number,
    schoolPeriodId: number,
    shifts: ShiftsForSemesters
}

export interface SchoolClassWithPlan {
    id: number | null,
    name: string,
    mainTeacher: UserView | null
    plan: Map<string, number>,
    shifts: ShiftsForSemesters
}


export interface ShiftsForSemesters {
    firstSemester: Shift,
    secondSemester: Shift,
}

export enum Shift {
    FIRST = "FIRST", SECOND = "SECOND"
}

export const shiftTranslation = {
    FIRST: "Първа смяна", SECOND: "Втора смяна"
}

File: backend/src/main/kotlin/com/nevexis/backend/schoolManagement/school_class/SchoolClass.kt
package com.nevexis.backend.schoolManagement.school_class

import com.nevexis.backend.schoolManagement.school_calendar.Shift
import com.nevexis.backend.schoolManagement.users.UserView
import kotlinx.serialization.Serializable

@Serializable
data class SchoolClass(
    val id: Int? = null,
    val name: String,
    val mainTeacher: UserView? = null,
    val schoolId: Int,
    val schoolPeriodId: Int,
    val shifts: ShiftsForSemesters
)

@Serializable
data class SchoolClassWithPlan(
    val id: Int? = null,
    val name: String,
    val mainTeacher: UserView? = null,
    val plan: Map<String, Int>,
    val shifts: ShiftsForSemesters
)

@Serializable
data class ShiftsForSemesters(
    val firstSemester: Shift,
    val secondSemester: Shift,
)